server-spi/src/main/java/org/keycloak/storage/user/UserQueryProvider.java

```diff
@@ -173,15 +173,16 @@ default int getUsersCount(RealmModel realm, boolean includeServiceAccount) {
     }
 
     /**
-     * Search for users with username, email or first + last name that is like search string.
+     * Searches for users with username, email or first + last name that is like search string.
```

Comment:
Should we do searching like described? As far as I know our implementations doesn't search for all words in `search` separately. The way I understand it, it just checks if any of strings `username`, `email` and `first name + last name` contains `search` string. See https://github.com/keycloak/keycloak/blob/master/model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java#L824-L833. Am I missing something?

Reply:
Yes. Building any sort of index on "first name + last" is hard and requires functional indices. It's easier to search word for word. JPA implementation is far from performant just because of this. For the fields alone, full-text search / index could be leveraged.